### DISCUSSION TITLE
Out of memory exception

### DISCUSSION THREAD

**Discussion by wsla1** [microsoft/onnxruntime#18704] [Other Q&A]
I'm trying to use Python ONNX Runtime in AWS Lambda (10 GB ram), to run a small-ish neural net - ONNX converted model is 319 KB.
When running inference, I'm getting out of memory exceptions - "Failed to allocate memory for requested buffer of size 7993016064"
What could I do to reduce the memory use? I went through the docs, but didn't find anything that would massively reduce the memory use

**Comment by Oil3** [comment]
Try the malloc alternative memory management? You can load it as an environment variable.
Does that talks to you or should I explain?
How about threading, can you use a sequencial/thread lock?
